src/services/logger.ts

```diff
@@ -18,49 +18,127 @@ if (config.nodeEnv !== 'dev' && config.nodeEnv !== 'test') {
   });
 }
 
-const logger = config.nodeEnv === 'dev' || config.nodeEnv === 'test' ? console : log4js.getLogger();
+const devEnv = config.nodeEnv === 'dev' || config.nodeEnv === 'test';
+const logger = devEnv ? console : log4js.getLogger();
+
+const createTimeStamp = (): string => {
+  const dateObj = new Date();
+
+  let date = dateObj.toISOString();
+  date = date.substring(0, date.indexOf('T'));
+
+  const time = dateObj.toLocaleTimeString([], { hour12: false });
+
+  return `${date} ${time}`;
+};
 
 export class LoggerService {
+  /*
+   * Internal fields are called by the class when each respective method is called
+   *
+   * Each field is by default `null`, see `constructor()` for how each log level is set */
+  #info: (message: string, serviceOperation?: string) => void = () => null;
+  #debug: (message: string, serviceOperation?: string) => void = () => null;
+  #trace: (message: string, serviceOperation?: string) => void = () => null;
+  #warn: (message: string, serviceOperation?: string) => void = () => null;
+  #error: (message: string | Error, innerError?: unknown, serviceOperation?: string) => void = () => null;
+  constructor() {
+    // enable logging only if we're on a dev environment or apm logging is activated
+    if (config.apmLogging || devEnv) {
+      switch (config.logger.logstashLevel) {
```

Comment:
Please consider the following - default known log levels are always UPPERCASE, eg, WARN, ERROR, INFO etc, so just adding this toLowerCase since your cases check for lowercase already, means less chances for unforeseen behaviour. This method would also only be called once during startup, so the additional overhead is not a big deal. 
`switch (config.logger.logstashLevel.toLowerCase()) {`

Reply:
Speaking of case, it seems like lowercase was used overall in configuration, but uppercase in output.

I based it off of this:

https://github.com/frmscoe/frms-coe-lib/blob/d6fb3606f46e73473df51de30026b4b834df68af/src/config.ts#L27

I'd have expected this to be ` || 'INFO'`.

Your suggestion is a silver bullet because it makes it insensitive as well. For some of us who will use `LOGSTASH_LEVEL=tRaCe` just for the chaos lol